data_loaders/cifar_data_loader.py

```diff
@@ -0,0 +1,93 @@
+import os
+import pickle
+from cfg_loader import ConfigLoader
+import numpy as np
+import polars as pl
+import torch
+from base_data_loader import BaseDataLoader
+
+
+class CifarDataLoader(BaseDataLoader):
+    def __init__(self, data_loader_cfg):
+        super().__init__()
+        self.train_data = None
+        self.validation_data = None
+        self.test_data = None
+        self.data_df = None
+        self.cfg = data_loader_cfg.image
+        self.num_labels = self.get_num_labels()
+        assert self.cfg.enabled == True
+
+    def get_num_labels(self):
+
+        with open(self.cfg.annotation_map_file, "rb") as f:
+            data = pickle.load(f, encoding="latin1")
+        return len(data["label_names"])
+
+    def load_data(self):
+
+        files = os.listdir(self.cfg.data_path)
+        images = []
+        labels = []
+        names = []
+
+        for file in files:
+            if "batch" in file.strip("_"):
```

Comment:
You are looking for batch in dataset contents. But, CIFAR 100 does not have any batch in their file names. Then this won't work for CIFAR 100.